recipes/lora_finetune_distributed.py

```diff
@@ -80,20 +80,7 @@ def __init__(self, cfg: DictConfig) -> None:
                 "full fp16 training is not supported with this recipe. Please use bf16 or fp32 instead."
             )
 
-        # For CUDA devices, check if the HW supports bf16 if bf16 is specified.
-        if (
-            self._dtype == torch.bfloat16
-            and self._device != torch.device("cpu")
-            and not torch.cuda.is_bf16_supported()
-        ):
-            raise RuntimeError("Full bf16 training is not supported on this hardware.")
-
-        world_size, rank = utils.get_world_size_and_rank()
```

Comment:
Now that we have our distributed tests working as expected, I'm removing this constraint.